lib/marisa/keyset.cc

```diff
@@ -45,7 +45,7 @@ void Keyset::push_back(const char *str) {
   assert(size_ < SIZE_MAX);
   MARISA_THROW_IF(str == nullptr, std::invalid_argument);
 
-  std::size_t length = 0;
+  uint32_t length = 0;
   while (str[length] != '\0') {
     ++length;
   }
```

Comment:
I think it might be better to keep this as a size_t, then assert that it fits in a uint32_t (it doesn't matter in practice, but it's cleaner and aligns with what the rest of the code already does, e.g., in `key.h`).